pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -53,7 +53,16 @@ type Webhook struct {
 	namespaceSelector *metav1.LabelSelector
 	objectSelector    *metav1.LabelSelector
 	containerRegistry string
-	datadogConfig     config.Component
+
+	profilesJSON                 string
+	provider                     string
+	imageName                    string
+	imageTag                     string
+	isLangDetectEnabled          bool
+	isLangDetectReportingEnabled bool
+	isClusterAgentEnabled        bool
+	clusterAgentCmdPort          int
+	clusterAgentServiceName      string
```

Comment:
small nits
if you can put a comment that explains why we have them part of the struct

```suggestion
    // all these fields are use to store datadog agent config parameters
    // to avoid calling the config resolution each time the webhook received
    // a request because this resolution is expensive in term of CPU.
	profilesJSON                 string
	provider                     string
	imageName                    string
	imageTag                     string
	isLangDetectEnabled          bool
	isLangDetectReportingEnabled bool
	isClusterAgentEnabled        bool
	clusterAgentCmdPort          int
	clusterAgentServiceName      string
```